### PR TITLE
Sql: add single-param trunc

### PR DESCRIPTION
**Pull request by chaas**
Add single-parameter `trunc` sql function, which truncates toward zero to a whole number.
Needed for Tableau support https://github.com/MaterializeInc/materialize/issues/11714.

### Motivation

This PR adds a known-desirable feature.

### Tips for reviewer

Let me know if the testing is adequate. Tested locally and via `sqllogictest`. Also, if any additional documentation is required.
Also thoughts on using `Rounding::Down` and then `cx.round`? Seems like a clever way to use Numeric's built-in truncation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

Add sql function `trunc(x: N) -> N` that truncates `x` toward zero to a whole number.

### CODE REVIEW

File: test/sqllogictest/arithmetic.slt [hunk 1]
```diff
@@ -498,6 +498,79 @@ SELECT round(CAST (5.0 AS float), 3.0)
...
+
+query R
+SELECT trunc(1.5678)
+----
+1
+
+statement ok
+DROP TABLE nums
+
+query error
```
Comment (mjibson) [inline test/sqllogictest/arithmetic.slt:571]: this takes a third argument that is a partially line matching regex, something like: `query error No function matches`

File: test/sqllogictest/arithmetic.slt [hunk 2]
```diff
@@ -498,6 +498,79 @@ SELECT round(CAST (5.0 AS float), 3.0)
 query error Cannot call function round\(double precision, double precision\): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts
 SELECT round(CAST (5.0 AS float), CAST (3.0 AS float))
 
+query R
```
Comment (mjibson) [inline test/sqllogictest/arithmetic.slt:501]: Can you add tests for nan, inf, and -0?

File: src/pgrepr/src/oid.rs [hunk 3]
```diff
@@ -100,3 +100,4 @@ pub const TYPE_ANYCOMPATIBLEMAP_OID: u32 = 16_455;
 pub const FUNC_MAP_LENGTH_OID: u32 = 16_456;
 pub const FUNC_MZ_PANIC_OID: u32 = 16_457;
 pub const FUNC_MZ_VERSION_NUM_OID: u32 = 16_458;
+pub const FUNC_TRUNC_F32_OID: u32 = 16_459;
```
Comment (mjibson) [inline src/pgrepr/src/oid.rs:103]: Ooh a bonus trunc function that postgres doesn't have. Nice.

**Review by mjibson — APPROVED** [review]

**Review by jkosh44 — APPROVED** [review]
LGTM

File: test/sqllogictest/arithmetic.slt (hunk 2 above)
Comment (chaas) [reply test/sqllogictest/arithmetic.slt:501]: Done! Assuming this produces the desired output
```
query RRRR
SELECT trunc('NaN'::decimal), trunc('Inf'::float), trunc(0), trunc(-0)
----
NaN  inf  0  0
```
I confirmed that this is postgresql's behavior too.

File: test/sqllogictest/arithmetic.slt (hunk 1 above)
Comment (chaas) [reply test/sqllogictest/arithmetic.slt]: Good suggestion, thanks. Done

**Pull request merged**